com.playeveryware.eos/Runtime/Core/Config/SandboxId.cs

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2021 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using System.Text.RegularExpressions;
+    using System;
+
+    public struct SandboxId : IEquatable<SandboxId>
+    {
+        private string _value;
+        private const string PreProductionEnvironmentRegex = @"^p\-[a-zA-Z\d]{30}$";
+
+        public string Value
+        {
+            readonly get
+            {
+                return _value;
+            }
+            set
+            {
+                if (!Guid.TryParse(value, out _))
+                {
+                    if (!Regex.IsMatch(value, PreProductionEnvironmentRegex))
+                    {
+                        throw new ArgumentException(
+                            "Value for SandboxId must either be " +
+                            "parseable to a Guid, or it must start with a " +
+                            "lowercase 'p', followed by a dash and thirty " +
+                            "letter characters.");
+                    }
+                }
+
+                _value = value;
+            }
+        }
+
+        public readonly bool Equals(SandboxId other)
+        {
+            return _value == other._value;
```

Comment:
The reason `==` is used here is because two `SandboxId` structs might _both_ have `_value` field members that are `null`, in which case they should be considered equal.

Reply:
I think I'd still want to use .Equals, it can be significant. Perhaps you could have an if statement for "is _value == null && other._value == null".

Actually since _value is a string, could you use `String.IsNullOrEmpty` on both instead of null checking directly?

Reply:
The following statement:
```c#
_value == null && other._value == null
```
I believe will _always_ evaluate to the same value as:
```c#
_value == other._value
```
Same thing as 
```c#
string.IsNullOrEmpty(_value) && string.IsNullOrEmpty(other_.value)
```

I believe that all three lines will in all cases evaluate to the same `true` or `false` value. Is there an edge case I'm forgetting?

Reply:
Ah - I think you are correct here. I'm also correct that those three lines will always come to the same evaluation. The distinction is in how strings might be compared - and allows (I believe) for culture-specific considerations when evaluating string equivalency. I'll update the implementation accordingly. Interesting catch and a good reminder, thanks Mint! :)

Reply:
Okay I did more research and discovered that I was wrong - string comparison overrides the `==` operator to properly do the value comparison between the two. Several reputable sources lead me to believe the original implementation using the `==` operator is the correct approach.